src/node/utils.ts

```diff
@@ -1,11 +1,31 @@
 import { Readable } from "stream";
+import { ReadableStream } from "web-streams-polyfill/ponyfill/es2018";
 
-export function streamToString(stream: Readable): Promise<string> {
-  const chunks: any[] = [];
+export function toReadableStream(stream: Readable): ReadableStream {
+  return new ReadableStream({
+    start(controller) {
+      stream.pause();
+      stream.on("data", chunk => {
+        controller.enqueue(chunk);
+        stream.pause();
+      });
+      stream.on("end", () => controller.close());
+      stream.on("error", e => controller.error(e));
+    },
+    pull() {
+      stream.resume();
+    },
+    cancel() {
+      stream.pause();
+    },
+  });
+}
 
-  return new Promise((resolve, reject) => {
-    stream.on("data", chunk => chunks.push(chunk));
-    stream.on("error", reject);
-    stream.on("end", () => resolve(Buffer.concat(chunks).toString("utf8")));
+export function stringToReadableStream(content: string): ReadableStream {
+  return new ReadableStream({
+    start(controller) {
+      controller.enqueue(content);
```

Comment:
We have to find out how to store the content as byte array instead of storing it as it is.

Reply:
I didn't understand your concern to e honest. But if something does not work try to split the task, remove the feature from your PR and create another issue for that. So that we can merge this PR and continue the development.

Reply:
Apparently I was just panicking